enterprise/enrollment_api.py

```diff
@@ -37,3 +49,21 @@ def enroll_user_in_course(user, course_details, mode):
     """
     client = get_enrollment_api_client()
     return client.enrollment.post(dict(user=user, course_details=course_details, mode=mode))
+
+
+def enroll_user_in_course_locally(user, course_id, mode):
+    """
+    Enroll a user in a course, using local database methods.
+
+    This is only used in one place; in models.py. It is necessary
+    because the enrollment API can't enroll a user before it exists, and
+    the post_save signal for users results in a user not actually being
+    saved until after the API call is made.
+
+    In Django 1.9 and later, there's a transaction.on_commit hook that we can
+    use to create a callback. Once we're able to depend on having Django 1.9, we
+    can shift over to that, but for right now, we have to do it this way.
+    """
+    if CourseKey is None and CourseEnrollment is None:
```

Comment:
I agree with @haikuginger with the use of a class for namespacing.  However, if this module is becoming unwieldy due to the amount of methods, then we could consider splitting the functions into two modules, which I would be okay with.